dynatrace_event/stackstate_checks/dynatrace_event/dynatrace_event.py

```diff
@@ -0,0 +1,264 @@
+            if not self.should_stop_processing(events_processed):
+                nextCursor = events.get("nextCursor")
+                self.log.info("Next cursor value is {}".format(nextCursor))
+                if nextCursor:
+                    self.log.info("Next cursor is not null, so recursing again...")
+                    self.log.info("Total events processed : {}".format(events_processed))
+                    self.collect_events(events_processed, cursor=nextCursor)
+        else:
+            self.log.debug("Error in pulling the events : {}".format(events.get("error").get("message")))
+        # push the last received timestamp in memory for the instance
+        self.status.data[self.url] = events.get("to")
+
+    def event_limit_exceeded_condition(self, events_response):
+        """
+        Check if number of events between subsequent check runs exceed the `events_process_limit`
+        :return: boolean True or False
+        """
+        total_event_count = events_response.get("totalEventCount")
+        # if events processed last time and total event count exceeded the limit
+        if self.status.data.get(self.url) is not None and total_event_count > self.events_process_limit:
+            self.log.warning("Maximum event limit to process is {} but received total {} "
+                             "events".format(self.events_process_limit, total_event_count))
+            self.reset_run_state = True
+            return True
+
+    def send_filtered_events(self):
+        """
+        Method to filter the closed events and create the health event for open events
+        :return:
+        """
+        events = self.status.data.get(self.url + "/events")
+        if events:
+            for entityId in events.keys():
+                open_events = []
+                event_type_values = events.get(entityId)
+                if event_type_values:
+                    for event_type in event_type_values.keys():
+                        event_status = event_type_values.get(event_type).get("eventStatus")
+                        if event_status == "CLOSED":
+                            del self.status.data[self.url + "/events"][entityId][event_type]
+                        else:
+                            self.log.info("Appending an open event for entityID {}: {}".format(entityId, event_type_values[event_type]))
+                            open_events.append(event_type_values[event_type])
+                    if not self.status.data[self.url + "/events"][entityId]:
+                        del self.status.data[self.url + "/events"][entityId]
+                    if len(open_events) > 0:
+                        self.create_health_event(entityId, open_events)
+                else:
+                    del self.status.data[self.url + "/events"][entityId]
+        self.log.info("After filtering the closed events :- {}".format(self.status.data))
+        if self.reset_run_state:
```

Comment:
if we reset the run state, we should send `CLEAR` events for all entities that PREVIOUSLY had an open state. You should thus use the old state and then iterate of these events. 

e.g.
```
except EventLimitReachedException:
  self.log.exception(str(e))
  self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=self.tags, message=str(e))
  state = self.load_state() # my functional method
   # for each entity in the old state send a CLEAR 
   state.clear()
   state.persists()
```

Reply:
done.